### PR TITLE
Fix flaky TestDAP_console test.

### PR DESCRIPTION
**Pull request by mbucko**
Test Plan:
llvm-lit llvm-project/lldb/test/API/tools/lldb-dap/console/TestDAP_console.py

### CODE REVIEW

**Review by walter-erquinigo — APPROVED** [review]

**Comment by JDevlieghere** [comment]
Would this affect other calls to `collect_console` too? Do we usually look for a particular string? If so, would it make sense to change the function (rather than this particular call site) or have a little helper?

**Comment by mbucko** [comment]
> Would this affect other calls to `collect_console` too? Do we usually look for a particular string? If so, would it make sense to change the function (rather than this particular call site) or have a little helper?

The problem is that the console hasn't had enough time to collect the logs after the process was terminated. This fix will keep collecting the logs until the "exited with status..." log has been collected so that we can check it. The caller could specify 10.0 seconds timeout instead of 1.0 second, but this way the caller can exit earlier rather than forcefully waiting on the timeout.

**Comment by JDevlieghere** [comment]
> The problem is that the console hasn't had enough time to collect the logs after the process was terminated. This fix will keep collecting the logs until the "exited with status..." log has been collected so that we can check it. The caller could specify 10.0 seconds timeout instead of 1.0 second, but this way the caller can exit earlier rather than forcefully waiting on the timeout.

I wasn't suggesting to increase the timeout, but instead change the function to do exactly what you did here: have a smaller timeout and repeatedly check for a particular string, assuming that's a common pattern. The motivation being that other places where we do this probably also suffer from this, and it's just a matter of time before we see flakiness there too.

File: lldb/test/API/tools/lldb-dap/launch/TestDAP_launch.py [hunk 1]
```diff
@@ -467,5 +467,5 @@ def test_terminate_commands(self):
         # Once it's disconnected the console should contain the
         # "terminateCommands"
         self.dap_server.request_disconnect(terminateDebuggee=True)
-        output = self.collect_console(duration=1.0)
+        output = self.collect_console(timeout_secs=1.0)
```
Comment (walter-erquinigo) [inline lldb/test/API/tools/lldb-dap/launch/TestDAP_launch.py:470]: add `terminateCommands` to the pattern

File: lldb/test/API/tools/lldb-dap/launch/TestDAP_launch.py [hunk 2]
```diff
@@ -337,7 +337,7 @@ def test_commands(self):
         # Get output from the console. This should contain both the
         # "exitCommands" that were run after the second breakpoint was hit
         # and the "terminateCommands" due to the debugging session ending
-        output = self.collect_console(duration=1.0)
+        output = self.collect_console(timeout_secs=1.0)
```
Comment (walter-erquinigo) [inline lldb/test/API/tools/lldb-dap/launch/TestDAP_launch.py:340]: add `terminateCommands` to the pattern

File: lldb/test/API/tools/lldb-dap/commands/TestDAP_commands.py [hunk 3]
```diff
@@ -75,6 +75,6 @@ def test_command_directive_abort_on_error_attach_commands(self):
             attachCommands=["?!" + command_quiet, "!" + command_abort_on_error],
             expectFailure=True,
         )
-        full_output = self.collect_console(duration=1.0)
+        full_output = self.collect_console(timeout_secs=1.0)
```
Comment (walter-erquinigo) [inline lldb/test/API/tools/lldb-dap/commands/TestDAP_commands.py:78]: add `settings` to the pattern

File: lldb/test/API/tools/lldb-dap/commands/TestDAP_commands.py [hunk 4]
```diff
@@ -47,7 +47,7 @@ def do_test_abort_on_error(
             postRunCommands=commands if use_post_run_commands else None,
             expectFailure=True,
         )
-        full_output = self.collect_console(duration=1.0)
+        full_output = self.collect_console(timeout_secs=1.0)
```
Comment (walter-erquinigo) [inline lldb/test/API/tools/lldb-dap/commands/TestDAP_commands.py:50]: add `settings` to the pattern

File: lldb/test/API/tools/lldb-dap/attach/TestDAP_attach.py [hunk 5]
```diff
@@ -235,5 +235,5 @@ def test_terminate_commands(self):
         # Once it's disconnected the console should contain the
         # "terminateCommands"
         self.dap_server.request_disconnect(terminateDebuggee=True)
-        output = self.collect_console(duration=1.0)
+        output = self.collect_console(timeout_secs=1.0)
```
Comment (walter-erquinigo) [inline lldb/test/API/tools/lldb-dap/attach/TestDAP_attach.py:238]: add `terminateCommands` to the pattern

File: lldb/test/API/tools/lldb-dap/attach/TestDAP_attach.py [hunk 6]
```diff
@@ -200,7 +200,7 @@ def test_commands(self):
         # Get output from the console. This should contain both the
         # "exitCommands" that were run after the second breakpoint was hit
         # and the "terminateCommands" due to the debugging session ending
-        output = self.collect_console(duration=1.0)
+        output = self.collect_console(timeout_secs=1.0)
```
Comment (walter-erquinigo) [inline lldb/test/API/tools/lldb-dap/attach/TestDAP_attach.py:203]: add `terminateCommands` to the pattern

**Review by JDevlieghere — APPROVED** [review]
Thanks. LGTM!

**Comment by walter-erquinigo** [comment]
@mbucko , could you request commit access to Chris Latner? 

```
We grant commit access to contributors with a track record of submitting high quality patches. If you would like commit access, please send an email to [Chris](mailto:clattner%40llvm.org) with your GitHub username.
```

That would be very helpful in case you break the build bots and want to push a hot fix.

File: lldb/test/API/tools/lldb-dap/attach/TestDAP_attach.py [hunk 7]
```diff
@@ -200,7 +200,10 @@ def test_commands(self):
         # Get output from the console. This should contain both the
         # "exitCommands" that were run after the second breakpoint was hit
         # and the "terminateCommands" due to the debugging session ending
-        output = self.collect_console(duration=1.0)
+        output = self.collect_console(
+            timeout_secs=1.0,
```
Comment (adrian-prantl) [inline lldb/test/API/tools/lldb-dap/attach/TestDAP_attach.py:204]: These timeouts look like they're way too small for CI (should be more in the 10s of seconds range and under ASAN they should be multipled by a factor 10).

**Pull request merged**